### PR TITLE
[SPARK-3596][YARN]Support changing the yarn client monitor interval

### PR DESCRIPTION
**Pull request by Sephiroth-Lin**

### CODE REVIEW

**Comment by WangTaoTheTonic** [comment]
Jenkins, test this please.

File: yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnClientSchedulerBackend.scala [hunk 1]
```diff
@@ -125,6 +125,7 @@ private[spark] class YarnClientSchedulerBackend(
    */
   private def asyncMonitorApplication(): Unit = {
     assert(client != null && appId != null, "Application has not been submitted yet!")
+    val interval = conf.getLong("spark.yarn.client.progress.pollinterval", 1000)
```
Comment (sryza) [inline yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnClientSchedulerBackend.scala]: I would change this to something like spark.yarn.client.amPollInterval.

Comment (Sephiroth-Lin) [inline yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnClientSchedulerBackend.scala]: Thank you, but as it is the client to get the application report from the RM, so maybe "spark.yarn.client.progress.pollinterval" is better.

**Comment by tgravescs** [comment]
test this please

File: yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnClientSchedulerBackend.scala (hunk 1 above)
Comment (sryza) [inline yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnClientSchedulerBackend.scala]: My main concern is that we don't really have a notion of "progress" in Spark on YARN.  In MapReduce, progress traditionally referred to how far along the app was, but Spark doesn't provide a similar notion.  YARN itself does have a notion of app progress, which is not related to what's here.

**Comment by tgravescs** [comment]
Jenkins, test this please

File: yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnClientSchedulerBackend.scala (hunk 1 above)
Comment (Sephiroth-Lin) [inline yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnClientSchedulerBackend.scala]: Yeah, you are right. In PR #5305 I use the client.monitorApplication, then we can use "spark.yarn.report.interval" to changing the yarn client monitor interval. Thank you.

Comment (srowen) [inline yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnClientSchedulerBackend.scala]: OK, is this superseded by #5305 ? it sounds like you have two different changes related to SPARK-3596 and I'm wondering whether they are mutually exclusive, or one or both is preferred, or what.

Comment (Sephiroth-Lin) [inline yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnClientSchedulerBackend.scala]: OK

**Pull request closed without merging**